ts/WoltLabSuite/Core/Component/User/Follow.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Handles the user follow buttons.
+ *
+ * @author Marcel Werk
+ * @copyright 2001-2024 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @since 6.1
+ */
+
+import { prepareRequest } from "WoltLabSuite/Core/Ajax/Backend";
+import { getPhrase } from "WoltLabSuite/Core/Language";
+
+function toggleFollow(button: HTMLButtonElement): void {
+  if (button.dataset.following != "1") {
+    button.dataset.following = "1";
+    button.dataset.tooltip = getPhrase("wcf.user.button.unfollow");
+    button.querySelector("fa-icon")?.setIcon("circle-minus");
+    void prepareRequest(button.dataset.endpoint!)
+      .post({
+        action: "follow",
+      })
+      .fetchAsResponse();
+  } else {
+    button.dataset.following = "0";
+    button.dataset.tooltip = getPhrase("wcf.user.button.follow");
+    button.querySelector("fa-icon")?.setIcon("circle-plus");
+    void prepareRequest(button.dataset.endpoint!)
+      .post({
+        action: "unfollow",
+      })
+      .fetchAsResponse();
+  }
+}
+
+export function setup(): void {
+  document.querySelectorAll<HTMLButtonElement>(".jsFollowButton").forEach((button) => {
```

Comment:
First of all, `setup()` is never invoked, but could very well be dynamically invoked through a `whenFirstSeen()` call in `BootstrapFrontend`?

Second, I recommend using `wheneverFirstSeen()` here because it makes sure that any dynamically registered instances, such as through a popover, are still registered just fine.

Third: Using `.jsFollowButton` is unsafe because this is the previously used class name that did not have the `data-endpoint` set. It is wise to use a different class name or possibly even an attribute that also serves as the property for the endpoint:

```smarty
<button
    type="button"
    data-follow-user="{link controller='UserFollow' id=$targetUser->userID}{/link}"
>…</button>
```